src/ngraph_builder1.cc

```diff
@@ -0,0 +1,370 @@
+/*******************************************************************************
+ * Copyright 2017-2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#include "ngraph_builder1.h"
+#include "ngraph_translateops.h"
+
+using namespace std;
+namespace ng = ngraph;
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+Status Builder1::TranslateGraph(
+    const std::vector<TensorShape>& inputs,
+    const std::vector<const Tensor*>& static_input_map,
+    shared_ptr<ng::Function>& ng_function) {
+  TF_RETURN_IF_ERROR(Initialize());
+
+  vector<shared_ptr<ng::op::Parameter>> ng_parameter_list;
+  TF_RETURN_IF_ERROR(GetInputParams(inputs, tf_params, ng_parameter_list));
+
+  TF_RETURN_IF_ERROR(TranslateEachOp(tf_ops, static_input_map));
+
+  VectNg ng_result_list;
+  TF_RETURN_IF_ERROR(GetOutputNodes(tf_ret_vals, ng_result_list));
+
+  // Create the nGraph function.
+  try {
+    ng_function = make_shared<ng::Function>(ng_result_list, ng_parameter_list);
+  } catch (...) {
+    return errors::Unimplemented("Unable to create nGraph function");
+  }
```

Comment:
Adding a try-catch here. Currently this is a catch-all, but in case ngraph has specific error classes, we can provide different messages for each case.

Reply:
Added this